src/main/java/org/radarcns/android/device/AbstractDeviceManager.java

```diff
@@ -122,7 +125,13 @@ protected void registerDeviceAtReady() {
     protected <V extends SpecificRecord> void send(AvroTopic<ObservationKey, V> topic, V value) {
         ObservationKey key = deviceStatus.getId();
         if (key.getSourceId() != null) {
-            dataHandler.addMeasurement(topic, key, value);
+            try {
+                dataHandler.addMeasurement(topic, key, value);
+            } catch (IllegalArgumentException ex) {
+                Crashlytics.log("Cannot send measurement for " + getState().getId());
```

Comment:
There might in some cases be a lot of logging messages if there is some problem with the device and it only sends `NaN`s for example. But I think the only other way to do this would be to traverse all records again before sending a cache and omitting the ones that do not validate, throwing a warning once for a whole cache. But maybe this is negligible. What do you think?

Reply:
Hmm good point. I introduced a `didWarn` variable for a similar case where the source was not registered yet. In this case I thought it might be good to know the variability of denied messages; that it would be a sufficiently rare occurrence not to be a resource hog. I think the actual resolution should be that the plugins check that they don't submit `NaN` data, and this is just an indicator to update the plugin.

Reply:
Yes I agree, this can and should be implemented in the device plugins. I would still leave this though, as a definite fallback if a plugin does not implement it.